src/meta/src/hummock/hummock_manager.rs

```diff
@@ -261,18 +262,23 @@ where
                     .levels
                     .insert(compaction_group.group_id(), Levels { levels });
             }
-            if version_ids.is_empty() {
-                version_ids.insert(init_version.id);
-            }
-            let mut redo_state = init_version.clone();
-            if version_ids.contains(&init_version.id) {
-                init_version.insert(self.env.meta_store()).await?;
-                versioning_guard
-                    .hummock_versions
-                    .insert(init_version.id, init_version);
-            }
+            version_ids.insert(init_version.id);
+            init_version.insert(self.env.meta_store()).await?;
+        }
 
-            for (id, version_delta) in &hummock_version_deltas {
+        versioning_guard.checkpoint_version_id = *version_ids.first().unwrap();
+        let mut redo_state = HummockVersion::select(
```

Comment:
Since there is only one checkpoint-version, we do not store this data by concat `checkpoint_version_id`.

Reply:
> Since there is only one checkpoint-version, we do not store this data by concat `checkpoint_version_id`.

May I have some more explanation? thanks

Reply:
I means that we can rewrite it by
```rust
let mut redo_state = HummockVersion::select(self.env.meta_store()).await?.unwrap();
```

Reply:
> I means that we can rewrite it by
> 
> ```rust
> let mut redo_state = HummockVersion::select(self.env.meta_store()).await?.unwrap();
> ```

but `insert` interface of metadata model is wrapped and I can't modify `key()` of a HummockVersion

Reply:
> why? We only need to store `HummockVersion` here and we can modify `key()`.

`HummockVersion` has `key()` which returns its id

Reply:
```rust
impl MetadataModel for HummockVersion {
    type KeyType = String;
    type ProstType = HummockVersion;

    fn key(&self) -> risingwave_common::error::Result<Self::KeyType> {
        Ok("".to_string())
    }
}
```

By doing this we can store one less key value

Reply:
> ```rust
> ```rust
> impl MetadataModel for HummockVersion {
>     type KeyType = String;
>     type ProstType = HummockVersion;
> 
>     fn key(&self) -> risingwave_common::error::Result<Self::KeyType> {
>         Ok("".to_string())
>     }
> }
> ```
> 
> 
>     
>       
>     
> 
>       
>     
> 
>     
>   
> By doing this we can store one less key value
> ```

fixed